transport/src/main/java/io/netty/channel/embedded/EmbeddedChannelId.java

```diff
@@ -50,12 +50,12 @@ public int compareTo(ChannelId o) {
 
     @Override
     public int hashCode() {
-        return super.hashCode();
+        return 0;
     }
 
     @Override
     public boolean equals(Object obj) {
-        return super.equals(obj);
+        return obj instanceof EmbeddedChannelId;
     }
 
     @Override
```

Comment:
These changes are necessary to make those methods work correctly in the case where an EmbeddedChannelId is serialized and then deserialized. The deserialized instance will not be the exact same instance as the one in the INSTANCE field. See issue #2869.

If you still object, please explain why. I will remove the change if necessary, but EmbeddedChannelIdTest would also need to be removed as that test would fail with the unmodified code.